src/endpoints/dir/checkDirExists.ts

```diff
@@ -0,0 +1,35 @@
+import Elysia from 'elysia';
+import {isPathValid} from '../../utils/pathUtils';
+import {getConfig} from '../../utils/config';
+import {join} from 'path';
+import {checkIfDirectoryExists} from '../../utils/directoryUtils';
+import {dirNotExistsMsg, provideValidPathDirMsg} from '../../constants/commonResponses';
+
+export function addCheckDirExistsEndpoint(app: Elysia) {
+  return app.get('dir/exists', async ({query, set}) => {
+    // Verify that the path is valid
+    let relativePath = query.path ? (query.path as string) : null;
+    if (!isPathValid(relativePath)) {
+      set.status = 400;
+      return provideValidPathDirMsg;
+    }
+    relativePath = relativePath as string;
+
+    // Get the full path to the directory
+    const dirPath = join(getConfig().dataDir, relativePath);
```

Comment:
You can get the relative path and absolute path at the same time using `validateRelativePath`. This way we will avoid same checking logic in multiple spots:

```cs
// Performs validation of the `query.path`
const dirPathValidationResult = validateRelativePath(query.path);
if (dirPathValidationResult.isErr()) {
  set.status = 400;
  return provideValidPathDirMsg;
}
// Create variable to hold relative and absolute path
const dirPath = sourcePathValidationResult.value;

// Using the relative and absolute path:
let something = dirPath.relativePath;
let somethingElse = dirPath.absolutePath;
```

Reply:
I reused it. Though the absolutePath doesn't seems necessary for this endpoint, I just used the relativePath from the value.

Reply:
Looks like you're using the absolute path in line 19:

```cs
// Get the full path to the directory
const dirPath = join(getConfig().dataDir, relativePath);
```

Here's whot it looks like in `validateRealativePath`:
https://github.com/Xkonti/file-api/blob/bd3b6fbfd75e3aea70dae70f8f9d1470f91a2869/src/utils/pathUtils.ts#L27-L30